R/import_ebird.R

```diff
@@ -1,51 +1,113 @@
-#' tar archive into a parquet-based database in your [`ebird_data_dir()`]
+#' @param tarfile path to the tar archive file downloaded from the eBird
+#'   website. Files containing either observation data (e.g.
+#'   `ebd_rel<DATE>.tar`) or checklist (e.g. `ebd_sampling_rel<DATE>.tar`) data
+#'   can be provided
+#'   
+#' @details 
+#' [eBird](https://ebird.org/home) data are collected and organized around the
+#' concept of a checklist, representing observations from a single birding
+#' event. Each checklist contains a list of species observed, counts of the
+#' number of individuals seen of each species, the location and time of the
+#' observations, and a measure of the effort expended while collecting these
+#' data. The majority of the [eBird](https://ebird.org/home) database is
+#' available for download in the form of the [eBird Basic Dataset
+#' (EBD)](https://ebird.org/data/download), a set of two tab-separated text
+#' files. 
 #' 
-#' @param tarfile path to the copy of `ebd_rel<DATE>.tar` file you downloaded
+#' The **checklist** dataset (referred to as the Sampling Event Data on the
+#' eBird website) consists of one row for each eBird checklist and columns
+#' contain checklist-level information such as location, date, and search
+#' effort. The **observation** dataset consists of one row for each species
+#' observed on each checklist and columns contain checklist-level information
+#' such as number of individuals detected. This dataset also contains all
+#' checklist-level variables, duplicated for each species on the same checklist.
+#'
+#' After [submitting a request for data
+#' access](https://ebird.org/data/download), users can download either or both
+#' of these datasets as tar archive files. `import_ebird()` takes the path to a
+#' tar file as input and imports the text file contained within to a parquet
+#' file, which will allow much easier access to the data. This function will
+#' automatically detect whether you are importing a checklist or observation
+#' dataset provided you **do not change the name of the downloaded file or
+#' unarchive the tar file**. The parquet files will be stored in the directory
+#' specified by [ebird_data_dir()], consult the help for that function to learn
+#' how to modify the parquet directory.
 #' 
+#' @return Invisibly return the path to the directory containing eBird parquet
+#'   files.
 #' @export
 #' @examples
+#' # only use a tempdir for this example, don't copy this for real data
+#' temp_dir <- file.path(tempdir(), "birddb")
+#' Sys.setenv("BIRDDB_HOME" = temp_dir)
 #' 
-#' Sys.setenv("BIRDDB_HOME"=tempdir())
-#' tar <- ebird_sample_data()
+#' # get the path to a sample dataset provided with the package
+#' tar <- sample_observation_data()
+#' # import the sample dataset to parquet
 #' import_ebird(tar)
 #' 
+#' unlink(temp_dir, recursive = TRUE)
 import_ebird <- function(tarfile) {
+  file_metadata <- parse_ebd_filename(tarfile)
```

Comment:
I don't think we use `file_metadata` anywhere except when it gets written out, right? Getting pedantic here, but seems a but odd to create this object now and ignore it until later when we write the file?

The code might be a tad cleaner to just have a function `record_metadata(ebd_file, dest)` that takes the filename as input, parses it, and writes the output.  that abstracts the details of how metadata recording is handled to a single fn, and the `import_ebird()` doesn't need to care about the mechanics.